xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java

```diff
@@ -66,4 +67,183 @@ public void testListServices() {
 		assertEquals("MemberList size", 3, result.getMemberList().getMember().size());
 	}
 
+	@Test
+	public void testIsSoapService() {
+		IsSoapService request = new IsSoapService();
+		request.setServiceCode("testService");
+		request.setSubsystemCode("TestSubSystem");
+		IsSoapServiceResponse result = (IsSoapServiceResponse)new WebServiceTemplate(marshaller).marshalSendAndReceive(
+				"http://localhost:" + port + "/ws/IsSoapService/", request);
+		assertNotNull(result);
+		assertEquals("Is given service a SOAP service", true, result.isSoap());
+	}
+
+	@Test
+	public void testIsSoapServiceFalse() {
+		IsSoapService request = new IsSoapService();
+		request.setServiceCode("getRandom");
+		request.setSubsystemCode("TestSubSystem");
+		IsSoapServiceResponse result = (IsSoapServiceResponse)new WebServiceTemplate(marshaller).marshalSendAndReceive(
+				"http://localhost:" + port + "/ws/IsSoapService/", request);
+		assertNotNull(result);
+		assertEquals("Is given service a SOAP service", false, result.isSoap());
+	}
+
+	@Test
+	public void testIsSoapServiceException() {
+		boolean thrown = false;
+		String exceptionMessage = null;
+		try {
+			IsSoapService request = new IsSoapService();
+			request.setServiceCode("testService123");
+			request.setSubsystemCode("TestSubSystem");
+			new WebServiceTemplate(marshaller).marshalSendAndReceive(
+					"http://localhost:" + port + "/ws/IsSoapService/", request);
+		} catch (SoapFaultClientException e) {
```

Comment:
The execptions could also be asserted without try-catch just having the test annotation like this
@Test(expected = SoapFaultClientException.class)

Reply:
This method does not provide a way to verify exception message